### PR TITLE
[Tests-Only] Add UI tests for settings UI

### LINKED ISSUE #686: Run UI tests of settings extension
The UI tests of the accounts extension are already executed in drone, lets do the same with the settings tests

### PR DESCRIPTION
**Pull request by micbar**
## Description

- Bring back the Settings Ui Tests pipeline
- Needs a phoenix release and a rebase to become green

closes #686 

@LukasHirt

### CODE REVIEW

**Comment by micbar** [comment]
The translations are outdated. 

This PR will be green after a new phoenix update.

**Review by individual-it — COMMENTED** [review]
should we add a retry analog to https://github.com/owncloud/ocis/pull/685/commits/253a4bdbc73a8817dd5f62ba6ab147b22f2ade9c ?

**Comment by dpakach** [comment]
@micbar I fixed the tests so they currently pass, maybe we can merge this now and when a new release of phoenix comes, we can fix the tests then.

**Comment by phil-davis** [comment]
Issue #697 is for the translation problem with menu items. Linking here so that when that issue is fixed, people can easily find this PR, see the diffs and know what kind of changes to revert.

**Review by phil-davis — APPROVED** [review]
LGTM. IMO get this in and running so that at least there will not be any more regressions.

**Comment by phil-davis** [comment]
> I fixed the tests so they currently pass
@dpakach not quite yet - https://drone.owncloud.com/owncloud/ocis/620/33/6
There are some fails like: 
```
   ✖ And the account menu should be listed in language "Deutsch" # ui/tests/acceptance/stepDefinitions/settingsContext.js:27
       AssertionError [ERR_ASSERTION]: the menu list were not same
           + expected - actual

            [
              "Profile"
              "Settings"
           +  "Accounts"
              "Abmelden"
            ]
```

**Comment by micbar** [comment]
@dpakach I would rather like to have thee final translation like i had it in my previous commit.

I extracted the values from the already existing translation on phoenix master branch.

**Comment by dpakach** [comment]
@micbar  done!

**Comment by phil-davis** [comment]
@micbar @dpakach where is this up to? Who needs to work on it next?

**Comment by micbar** [comment]
Currently, the translations are not matching.

We need to find out, if the problem is real.

**Comment by kulmann** [comment]
Rebased to current master and adapted drone config for settings. Let's see what CI says...

**Comment by individual-it** [comment]
it does not find the general UI test steps

image

**Comment by individual-it** [comment]
I cannot see how we get data into `TEST_INFRA_DIRECTORY`

File: accounts/ui/tests/run-acceptance-test.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ trap clean_up SIGHUP SIGINT SIGTERM
 if [ -z "$TEST_INFRA_DIRECTORY" ]
 then
 	cleanup=true
-	testFolder=$(< /dev/urandom tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+	testFolder=$(mktemp -d)
 	printf "creating folder $testFolder for Test infrastructure setup\n\n"
 	export TEST_INFRA_DIRECTORY=$testFolder
```
Comment (dpakach) [inline accounts/ui/tests/run-acceptance-test.sh:34]: this should be `$testFolder/tests` since we are now copying to a temp dir that's already created.

**Comment by dpakach** [comment]
@kulmann I have rebased this Pr and fixed the error on settings UI tests, this should now the UI tests from settings extension, Let's see if it passes on the CI now

**Comment by kulmann** [comment]
Mhm... still failing
Thank you @dpakach for taking care of this!

FYI, I changed test scenarios so that they always revert the language to `English` at the end. Also, The language switch needs to take effect without page reload. Because of that I added steps for checking the changed language without page reload as well.

**Comment by refs** [comment]
ping @micbar status?

**Comment by micbar** [comment]
The test PR is open and demonstrates a misbehavior of the WebUI

@kulmann

**Comment by kulmann** [comment]
I will rebase the branch. There have been some fixes regarding settings lately. Maybe test failures are (partially) fixed with that.

**Comment by dpakach** [comment]
One line was missing in the .drone.star file, I've added it

**Comment by phil-davis** [comment]
Tests failed 5 days ago: https://drone.owncloud.com/owncloud/ocis/2635/74/7
And now there are conflicts.

I requested review again from people so that they see this in GitHub Review Requests.

There is no assignee. Who is moving this forward?

**Comment by kulmann** [comment]
Assigned myself. Rebasing...

**Comment by phil-davis** [comment]
@kulmann @micbar this is on my GitHub review requests.
Is it progressing?
Do you want someone else to take this over and get it finished?

**Comment by kulmann** [comment]
> @kulmann @micbar this is on my GitHub review requests.
> Is it progressing?
> Do you want someone else to take this over and get it finished?

No progress. Would be awesome if someone from your team could take over. Thank you!

**Comment by phil-davis** [comment]
> No progress. Would be awesome if someone from your team could take over. Thank you!

We will assign somebody...

**Comment by dpakach** [comment]
Fixed the tests and updated everything, it should be passing in ci now. Please review
@kulmann @phil-davis @individual-it

**Review by kulmann — APPROVED** [review]

**Comment by kulmann** [comment]
Thank you for taking care @dpakach

**Pull request merged**